torch/distributed/_tensor/ops/utils.py

```diff
@@ -118,6 +118,22 @@ def is_tensor_shardable(shape: Sequence[int], spec: DTensorSpec) -> bool:
     return True
 
 
+def is_tensor_evenly_shardable(shape: Sequence[int], spec: DTensorSpec) -> bool:
+    """Check if the shape is evenly shardable according to the spec."""
+    # number of shards in each tensor dimension
+    shards_map = [1] * len(shape)
+    for i, placement in enumerate(spec.placements):
+        if placement.is_shard():
+            shard_dim = cast(Shard, placement).dim
+            shards_map[shard_dim] *= spec.mesh.size(i)
+
+    for i, dim_size in enumerate(shape):
+        if shards_map[i] > 1 and dim_size % shards_map[i]:
+            return False
```

Comment:
This is correct in C/C++ but not in Python:

```
>>> 5 % 5 == True
False
>>> 5 % 2 == True
True
>>> 5 % 3 == True
False
```

should be `(dim_size % shards_map[i] != 0)`